### ISSUE TITLE
Var Int Problem

### ISSUE THREAD

**Issue by p4skal** [PaperMC/Waterfall#426]
Our Waterfall recently had problems decoding var ints. It was possible that an attacker could send bytes all matching 0x80 bitmask. The cosequence is that it costs really much cpu time.

Old Code: 
`public static int readVarIntOLD(final ByteBuf input, final int maxBytes) {

      int out = 0;
      int bytes = 0;
      byte in;
      do {
          in = input.readByte();
          out |= (in & 0x7F) << bytes++ * 7;
          if (bytes > maxBytes) {
              throw new RuntimeException("VarInt too big");
          }
      } while ((in & 0x80) == 0x80);
      return out;
  }`

New Code: `public static int readVarInt(ByteBuf input, int maxBytes) {
    int out = 0;
    int bytes = 0;
    byte in;
    while (true) {
      in = input.readByte();

      out |= (in & 0x7F) << (bytes++ * 7);

      if (bytes > maxBytes) {
        throw new RuntimeException("VarInt too big");
      }

      if ((in & 0x80) != 0x80) {
        break;
      }
    }

    return out;
  }`

**Comment by mikroskeem** [comment]
1) What are you trying to say exactly?
2) Code blocks are hardly understandable because of formatting issues... besides, `New code` one is equal to the one in current DefinedPacket class?

**Comment by Janmm14** [comment]
old and new code are likely nearly equally translated to bytecode, logic seems identical too
@p4skal

**Comment by MrIvanPlays** [comment]
`DefinedPacket.readVarInt(<netty ByteBuf>)`

Done

**Comment by p4skal** [comment]
Sorry for the late response, i was on vacation.

@Janmm14 Your completly right, even from the logic perspective it's identical. 

We had profiled our waterfall fork and found out, that there was much time spent in the readVarInt method. So I looked online for a solution and found this code.I seem to just have copy & pasted it, created an issue without analysing it properly. I just wanted to fix it before vacation. 
The "old code" is actually from our fork, so it has nothing to do with the original bungee / waterfall code. Till today, there was no issue like this anymore.
Thank you for your effort and quick responses!
